src/identification.js

```diff
@@ -0,0 +1,109 @@
+"use strict";
+
+const fs = require("fs");
+const net = require("net");
+const colors = require("colors/safe");
+const Helper = require("./helper");
+
+class Identification {
+	constructor(startedCallback) {
+		this.connectionId = 0;
+		this.connections = new Map();
+
+		if (typeof Helper.config.oidentd === "string") {
+			this.oidentdFile = Helper.expandHome(Helper.config.oidentd);
+			log.info(`Oidentd file: ${colors.green(this.oidentdFile)}`);
+
+			this.Refresh();
+		}
+
+		if (Helper.config.identd.enable) {
+			if (this.oidentdFile) {
+				log.warn("Using both identd and oidentd at the same time, this is most likely not intended.");
+			}
+
+			var server = net.createServer(this.ServerConnection.bind(this));
+			server.listen({
+				port: Helper.config.identd.port || 113,
+				host: Helper.config.host,
+			}, () => {
+				var address = server.address();
+				log.info(`Identd server available on ${colors.green(address.address + ":" + address.port)}`);
+
+				startedCallback();
+			});
+		} else {
+			startedCallback();
+		}
+	}
+
+	ServerConnection(socket) {
+		socket.on("data", (data) => {
+			this.RespondToIdent(socket, data);
+			socket.end();
+		});
+	}
+
+	RespondToIdent(socket, data) {
+		data = data.toString();
+		data = data.split(",");
+
+		const lport = parseInt(data[0]);
+		const fport = parseInt(data[1]);
+
+		if (lport < 1 || fport < 1 || lport > 65535 || fport > 65535) {
+			return;
+		}
+
+		for (var connection of this.connections.values()) {
+			if (connection.socket.remoteAddress === socket.remoteAddress
+			&& connection.socket.remotePort === fport
+			&& connection.socket.localPort === lport
+			&& connection.socket.localAddress === socket.localAddress) {
+				return socket.write(`${lport}, ${fport} : USERID : UNIX : ${connection.user}\r\n`);
+			}
+		}
+
+		socket.write(`${lport}, ${fport} : ERROR : NO-USER\r\n`);
+	}
+
+	AddSocket(socket, user) {
+		const id = this.connectionId++;
```

Comment:
Is that a problem outside of exceeding maximum integer (which is [rather high](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER))?

Could do something like http://stackoverflow.com/a/3439649/1935861 where you store the deleted ids in an array and when you add a socket, you pick one of them (first, last, w/e) and if none, increase counter. That seems overkill to me, but I don't know enough about the overall feature to know if it's a DoS vector or not (if not, no need to care IMO).